pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go

```diff
@@ -836,10 +836,13 @@ func (t *Test) BackgroundCommand(
 // synchronously as a regular startup function. `runCmd` is the
 // command to actually run the command; it is run in the background.
 //
-// If overrideBinary is true, the binary used to run the command(s) will
-// be replaced with the cockroach binary of the current version the
-// cluster is running in.
-// TODO(testeng): Replace with https://github.com/cockroachdb/cockroach/issues/147374
+// By default, the binary used to run the command(s) will be the same as the
+// the current version of the cluster at the time this hook is executed.
+// This is because the binary version is no longer backwards compatible as of
+// v25.3
+//
+// If overrideBinary is true, the binary used to run the command(s) will be
```

Comment:
This bool sounds like it needs to be flipped to make sense. If it's true and we use the binary passed in the command, then it sounds like we aren't overriding anything at all.